### PR TITLE
rewrite to use concurrency

### PR DESCRIPTION
**Pull request by AmilKey**
latest ember ember-cli
ember-concurrency
tests

@knownasilya @medokin  ^^

### CODE REVIEW

**Comment by knownasilya** [comment]
Quite a big PR. I would have preferred a smaller PR with the features you wanted and another with the ember-cli update. It'll take a bit to sort everything out.

**Comment by medokin** [comment]
Looks fine, and I agree with @knownasilya that small PR's are way better to handle, but in this case it just makes sense to update the cli.

The style change on the other hand, belongs in a seperate PR.

File: app/components/new-version-notifier.js [hunk 1]
```diff
@@ -1,10 +1 @@
...
-
-import config from '../config/environment';
-import NewVersionNotifier from 'ember-cli-new-version/components/new-version-notifier/component';
-
-let versionFileName = config.newVersion ? `/${config.newVersion.fileName}` : '/VERSION.txt';
-
-export default NewVersionNotifier.extend({
-  versionFileName
-});
+export { default } from 'ember-cli-new-version/components/new-version-notifier/component';
```
Comment (knownasilya) [inline app/components/new-version-notifier.js]: How are you passing down the configuration now? Looks like this was missed.

File: ember-cli-build.js [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-/* global require, module */
-var EmberAddon = require('ember-cli/lib/broccoli/ember-addon');
+/* eslint-env node */
+const EmberAddon = require('ember-cli/lib/broccoli/ember-addon');
 
 module.exports = function(defaults) {
   var app = new EmberAddon(defaults, {
     // Add options here
     newVersion: {
-      fileName: 'MY-VERSION.txt'
```
Comment (knownasilya) [inline ember-cli-build.js]: Should this have been removed?

File: index.js [hunk 3]
```diff
@@ -9,20 +9,21 @@ module.exports = {
   /*
   Store config from `ember-cli-build.js`
    */
-  included: function(app, parentAddon) {
+  included: function(app/*, parentAddon*/) {
     this.options = app.options.newVersion || {};
 
-    if (this.options === true) {
-      this.options = { fileName: 'VERSION.txt' };
+    if (this.options.enabled === true) {
```
Comment (knownasilya) [inline index.js]: Same as above. This should be coming through via the configuration.

File: package.json [hunk 4]
```diff
@@ -2,56 +2,54 @@
   "name": "ember-cli-new-version",
   "version": "1.0.8",
   "description": "A convention based update notification for Ember. With this addon, you can detect a new version and notify the user to refresh the page",
+  "keywords": [
+    "ember-addon"
```
Comment (knownasilya) [inline package.json]: Keywords are missing from merge.

File: tests/integration/components/new-version-notifier-test.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    ++callCount;
+
+    return callCount < 4 ? 'v1.0.' + callCount : 'v1.0.3';
+  });
+
+  this.render(hbs`{{new-version-notifier updateInterval=100 version=version lastVersion=lastVersion}}`);
+
+  const done = assert.async(3);
+
+  later(() => {
```
Comment (knownasilya) [inline tests/integration/components/new-version-notifier-test.js]: The `wait` helper might be the better option here. Unless you have a specific reason not to use it?

File: app/components/new-version-notifier.js (hunk 1 above)
Comment (AmilKey) [inline app/components/new-version-notifier.js]: I use config from ember-cli-build.js

File: ember-cli-build.js (hunk 2 above)
Comment (AmilKey) [inline ember-cli-build.js]: yes, I use `VERSION.txt` as default name
https://github.com/sethwebster/ember-cli-new-version/pull/34/files/a4f8570c8395bb197df465fc3e72bee0b4daf90d#diff-168726dbe96b3ce427e7fedce31bb0bcR16

File: index.js (hunk 3 above)
Comment (AmilKey) [inline index.js]: I use config from ember-cli-build.js

File: tests/integration/components/new-version-notifier-test.js (hunk 5 above)
Comment (AmilKey) [inline tests/integration/components/new-version-notifier-test.js]: I prefer assert.async

File: index.js (hunk 3 above)
Comment (knownasilya) [inline index.js]: I'd rather not change that since that would require a major release, which adds little benefit for the user at this point.

Comment (AmilKey) [inline index.js]: I have added one feature that named `prepend` it requires a prepend variable that is used in ember-cli-build file

Comment (knownasilya) [inline index.js]: What does this feature do, and why is it required to be in ember-cli-build?

Comment (AmilKey) [inline index.js]: I use this config in my projects, so I need `prepend` url for getting `prepend/VERSION.txt`
```
    fingerprint: {
      prepend: process.env.PREPEND
    }
```

**Comment by sethwebster** [comment]
Any progress on changes, @AmilKey?

**Comment by AmilKey** [comment]
@sethwebster I use my fork in production and it works fine

**Comment by sethwebster** [comment]
@AmilKey Are you planning to resolve conflicts and commit upstream?

**Comment by AmilKey** [comment]
@sethwebster my changes make sense?

**Pull request merged**